lib/get-seam.ts

```diff
@@ -1,25 +1,31 @@
-import { SeamHttp } from "@seamapi/http/connect"
+import { SeamHttp, SeamHttpMultiWorkspace } from "@seamapi/http/connect"
 import { getConfigStore } from "./get-config-store"
 import { getServer } from "./get-server"
 
-export const getSeam = () => {
+export const getSeam = async (): Promise<SeamHttp> => {
   const config = getConfigStore()
 
   const token = config.get(`${getServer()}.pat`)
-  const token_type = token.startsWith("seam_at")
-    ? "pat"
-    : token.startsWith("ey")
-    ? "console-session-token"
-    : "api-key"
-
-  return new SeamHttp({
-    endpoint: getServer(),
-
-    personalAccessToken: token_type === "pat" ? token : undefined,
-    consoleSessionToken:
-      token_type === "console-session-token" ? token : undefined,
-    apiKey: token_type === "api-key" ? token : undefined,
-    // https://github.com/seamapi/javascript-http/issues/30
-    workspaceId: config.get("current_workspace_id"),
-  })
+
+  const workspaceId = config.get("current_workspace_id")
+
+  const options = { endpoint: getServer() }
+
+  if (token.startsWith("seam_at")) {
```

Comment:
This logic is contained in the SDK but not exported. I wonder if we should export `isPersonalAccessToken` etc? Or `detectTokenFormat` there?

Reply:
Honestly I would also recommend SeamHttp.fromToken, I keep reimplementing this and it's error prone logic, it took me a couple releases and people reporting bugs to handle every case

Reply:
I can support `SeamHttp.fromToken` but how to deal with the workspaceId or other "required" params that vary based on the token?

Reply:
(Merging this for now but if we can get something upstream to handle it we can use it here)